toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -672,6 +687,16 @@ func (g *PkgGraph) AllTestNodes() []*PkgNode {
 	})
 }
 
+// AllImplicitNodes returns a list of all implicit remote nodes in the graph
+func (g *PkgGraph) AllImplicitNodes() []*PkgNode {
+	return g.allNodesOfType(func(n *LookupNode) *PkgNode {
```

Comment:
just noticed that we have 2 functions in this file, `allNodesOfType`, and `NodesMatchingFilter` which do similar thing. Can we just keep one of them, preferably `NodesMatchingFilter` since the name is more descriptive of a generic function.

Reply:
They are similar, but slightly different. `allNodesOfType()` operates on the internal graph lookup table, which indexes based on `pkgNode.VersionedPkg`. Ideally, we don't ever want to expose the lookup table externally. For the three basic All `AllBuildNodes()`, `AllTestNodes()`, `AllImplicitNodes()` I think I can convert them over, but I'll have to think on `AllPreferredRunNodes()`.

Reply:
Ok, I was able to make it work